aptos-labs/aptos-core
# [tf][aws] update to latest calico and use gp3

### Description

Bring our EKS setup up-to-date with the latest recommendations for Kubernetes 1.22:

#### Calico
* Install Calico using the official `tiger-operator` helm chart, and delete the old one checked into our repo. The old helm chart was not compatible with k8s 1.22 and needed to be deprecated

#### ALB Ingress Controller
* Upgrade to the latest version
* Install it as a standalone helm chart
* Grant `k8s-aws-integrations` role/serviceaccount an additional permission that is required in the latest version
* Remove trailing wildcard (i.e. `/*`) in ingress paths, as that is no longer supported

#### External DNS
* Upgrade to the latest version
* Install it as a standalone helm chart

#### EKS
* Install EKS NodeGroup taints via the AWS Terraform Provider rather than patching in using a user_data script. This removes additional complexity
* Install EBS CSI driver and use GP3 by default. This is done as an EKS Addon, see:
  * https://docs.aws.amazon.com/eks/latest/userguide/ebs-csi.html
  * https://docs.aws.amazon.com/eks/latest/userguide/managing-ebs-csi.html
  * Note: this PR does not provide instructions for migrating from GP2 to GP3. During TF apply, operators may see that PVCs are immutable and that you cannot patch from GP2 to GP3. There are guides online on how to migrate the underlying EBS volume across storage classes: https://aws.amazon.com/blogs/containers/migrating-amazon-eks-clusters-from-gp2-to-gp3-ebs-volumes/. If the data is not critical though, the fastest way is to just delete these existing volumes and have k8s re-create them as GP3 volumes. If it's blockchain data (e.g. for a testnet), the chain will restart and sync back up

### Test Plan

TF apply in a new testnet. Check that calico NetworkPolicies still work:
* allow: network is live
* block: spin up a test pod and try to access the validator service. it gets blocked

```
kubectl -n default run -i --tty busybox --image=busybox:1.28 -- sh

# then in the pod, try to hit the validator directly, which fails due to the NetworkPolicy
/ # wget aptos-node-2-validator:8080
Connecting to aptos-node-2-validator:8080 (10.100.194.228:8080)
... spin and then fail to connect...
```

Check that we can create GP3 volumes:
* StorageClass created:
```
$ kubectl get sc
NAME   PROVISIONER       RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
gp3    ebs.csi.aws.com   Delete          WaitForFirstConsumer   true                   42m
```

* New volumes from a clean TF apply are GP3
```
$ kubectl get pvc
NAME                                         STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
aptos-node-0-validator-e5                    Bound    pvc-3cf038e0-1b5c-4624-9344-92986228830e   350Gi      RWO            gp3            26m
aptos-node-1-validator-e5                    Bound    pvc-7faf3d60-df4e-470e-a7fc-d1b27b82e4dc   350Gi      RWO            gp3            26m
aptos-node-2-validator-e5                    Bound    pvc-4a1ee6fa-4d30-4931-bd2b-46a3bd7d5773   350Gi      RWO            gp3            26m
aptos-node-mon-aptos-monitoring-prometheus   Bound    pvc-e100b6b1-163e-4d08-91b9-fe9fc5ceab64   100Gi      RWO            gp3            27m
fn-aptos-node-0-fullnode-e5-0                Bound    pvc-dde720a1-e2f2-48de-84fb-295ea9c9e6e3   350Gi      RWO            gp3            25m
fn-aptos-node-1-fullnode-e5-0                Bound    pvc-b093f0a7-89a8-4afe-a954-d42ebe195d82   350Gi      RWO            gp3            25m
```




- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/aptos-labs/aptos-core/2669)
